src/shared/util.js

```diff
@@ -1026,6 +1026,16 @@ function createPromiseCapability() {
   return capability;
 }
 
+function getNormalizeRegex() {
+  // In order to generate the following regex:
+  //  - create a PDF containing all the chars in the range 0000-FFFF with
+  //    a NFKC which is different of the char.
+  //  - copy and paste all those chars and get the ones where NFKC is
+  //    required.
+  // It appears that most the chars here contain some ligatures.
+  return /[\u00a0\u00b5\u037e\u0eb3\u2000-\u200a\u202f\u2126\ufb00-\ufb04\ufb06\ufb20-\ufb36\ufb38-\ufb3c\ufb3e\ufb40-\ufb41\ufb43-\ufb44\ufb46-\ufba1\ufba4-\ufba9\ufbae-\ufbb1\ufbd3-\ufbdc\ufbde-\ufbe7\ufbea-\ufbf8\ufbfc-\ufbfd\ufc00-\ufc5d\ufc64-\ufcf1\ufcf5-\ufd3d\ufd88\ufdf4\ufdfa-\ufdfb\ufe71\ufe77\ufe79\ufe7b\ufe7d]+/gu;
+}
+
```

Comment:
Would it perhaps make sense to just expose a helper function that does the normalization instead, rather than only the regular expressions, e.g. something like the following since that'd slightly simplify both call-sites?
```suggestion
let NormalizeRegex;

function normalizeUnicode(str) {
  if (!NormalizeRegex) {
    // In order to generate the following regex:
    //  - create a PDF containing all the chars in the range 0000-FFFF with
    //    a NFKC which is different of the char.
    //  - copy and paste all those chars and get the ones where NFKC is
    //    required.
    // It appears that most the chars here contain some ligatures.
    NormalizeRegex = /[\u00a0\u00b5\u037e\u0eb3\u2000-\u200a\u202f\u2126\ufb00-\ufb04\ufb06\ufb20-\ufb36\ufb38-\ufb3c\ufb3e\ufb40-\ufb41\ufb43-\ufb44\ufb46-\ufba1\ufba4-\ufba9\ufbae-\ufbb1\ufbd3-\ufbdc\ufbde-\ufbe7\ufbea-\ufbf8\ufbfc-\ufbfd\ufc00-\ufc5d\ufc64-\ufcf1\ufcf5-\ufd3d\ufd88\ufdf4\ufdfa-\ufdfb\ufe71\ufe77\ufe79\ufe7b\ufe7d]+/gu;
  }
  return str.replaceAll(NormalizeRegex, m => m[0].normalize("NFKC"));
}

```